### PR TITLE
feat: change RM Name to cluster name

### PR DESCRIPTION
**Pull request by amandavialva01**
## Ticket
DET-10310

## Description
Referring to the RM in the master config by its `Name` rather than its corresponding `ClusterName` can be slightly misguiding given that the rest of the config provided for the given resource manager is cluster-specific. 
Referring to an RM by its Name implies that the underlying determined resource manager works independently of its corresponding cluster (so changing the cluster for an RM may not prompt an RM name change). However, this is not the case. Change the `Name` key to `ClusterName` to increase emphasis on the necessity of a specific cluster being tied to a given RM.

NOTE: This is a breaking change.

## Test Plan
CI Passes.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by EmilyBonar — COMMENTED** [review]
There are two more locations on web where the name needs to be changed:

1. Lines 145 and 146 in `ResourcePoolCard.tsx`
2. Line 128 in `stores/cluster.tsx`

**Review by kkunapuli — CHANGES_REQUESTED** [review]
Looks great! A couple small requests.

With this change, `name` will be ignored if it is set, right?

I think we should add a bit of validation to warn the user, if `name` is set. I think `Deprecations()` function might be a good place? https://github.com/determined-ai/determined/blob/274d7638f242364db2d64c2c6cb734839450d526/master/internal/config/config.go#L395 

Also, please update docs when you're ready! Please add a release note and update the master config reference.

**Comment by channolanp** [comment]
When `additional_resource_managers[].resouce_manager.cluster_name` is omitted, the error message doesn't help lead the customer to the right solution:

> 	error found at root.ResourceConfig.AdditionalResourceManagersInternal[0].ResourceManager.KubernetesRM: name is required:  must be non-empty

That would make me think I should use `name` instead of `cluster_name`

**Review by EmilyBonar — APPROVED** [review]
Frontend side looks good!

**Review by kkunapuli — APPROVED** [review]
LGTM - nice work!

File: docs/release-notes/deprecate-name-use-cluster-name.rst [hunk 1]
```diff
@@ -0,0 +1,24 @@
+:orphan
+
+**Breaking Changes**
+
+-  Master: Add config field ``resource_manager.cluster_name`` and deprecate ``resource_anager.name``
```
Comment (tara-det-ai) [inline docs/release-notes/deprecate-name-use-cluster-name.rst:5]:
```suggestion
- Master: Add config field ``resource_manager.cluster_name`` and deprecate ``resource_manager.name``
```

File: docs/release-notes/deprecate-name-use-cluster-name.rst [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+            We change ``resource_manager.name`` to ``resource_manager.cluster_name`` to increase the
+            emphasis on the correlation between a given cluster's resource availability and the
+            Determined resource manager's behavior.
+
+      -  Since ``cluster_name`` is used as a unique identifier tied to your Determined resource
+            managers, this field must be unique for all resource managers when deploying multiRM
+            Determined.
+
+      -  When upgrading Determined, please replace ``name`` with ``cluster_name`` in the
+            ``resource_manager`` section of your master configuration YAML.
```
Comment (tara-det-ai) [inline docs/release-notes/deprecate-name-use-cluster-name.rst:24]: :orphan

**Breaking Changes**

-  Master Configuration:  The ``resource_manager.name`` field is replaced by ``resource_manager.cluster_name`` for better clarity and to support multiple resource managers.
- Resource managers operate relative to the specific cluster providing resources for Determined tasks. - Changing the cluster will affect the resource manager's responses.
- The ``cluster_name`` must be unique for all resource managers when deploying multiple resource managers in Determined.
- During upgrade, replace ``name`` with ``cluster_name`` in the ``resource_manager`` section of your master configuration YAML.

File: docs/reference/deploy/master-config-reference.rst [hunk 3]
```diff
@@ -247,12 +247,21 @@ The resource manager used to acquire resources. Defaults to ``agent``.
...
+================
+
+Optional for singleRM Determined. Required for multiRM Determined. Specifies the resource manager's
+associated cluster name. This references the cluster on which a Determined deployment is running.
+Defaults to ``default`` if not specified. For Kubernetes installations with additional resource
+managers, ensure unique names for all resource managers in the cluster.
+
+**NOTE:** ``resource_manager.cluster_name`` is separate from the ``cluster_name`` field of the
+master config that provides a readable name for the Determined deployment.
+
```
Comment (tara-det-ai) [inline docs/reference/deploy/master-config-reference.rst:260]: do we know if these are known acronyms for our users: "singleRM" and "multiRM"

if not, we'll need to spell them out, e.g.,

Optional for single resource manager configurations. Required for multiple resource manager configurations.

**Review by tara-det-ai — APPROVED** [review]
LGTM

**Pull request merged**